### PR TITLE
Enable pedantic analysis

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

**Review by ferndot — APPROVED** [review]
👍

File: pubspec.yaml [hunk 1]
```diff
@@ -43,6 +43,7 @@ dev_dependencies:
     sdk: flutter
   json_serializable: ^3.2.5
   file: ^5.1.0
+  pedantic: ^1.8.0
```
Comment (ferndot) [inline pubspec.yaml:46]: Non-blocker: why are we using 1.8 vs the latest version (1.9) here?

Comment (bkonyi) [reply pubspec.yaml:46]: Can't remember the specific package, but I'm 99% sure something in Flutter pins to 1.8.0. Either way, it caused version solving to fail.

Comment (dnfield) [reply pubspec.yaml:46]: Flutter is currently pinned to 1.8.0.

We could override that with a `dependency_override`

**Pull request merged**